packages/core/src/pipeline/config/graph/PipelineGraph.tsx

```diff
@@ -451,54 +589,9 @@ export class PipelineGraph extends React.Component<IPipelineGraphProps, IPipelin
   }
 
   public render() {
-    const { execution } = this.props;
-    const { allNodes, graphHeight, graphWidth, labelOffsetX, labelOffsetY, maxLabelWidth, nodeRadius } = this.state;
-
     return (
       <div className="pipeline-graph" ref={this.refCallback} onWheel={this.handleWheel}>
-        <svg
-          className="pipeline-graph"
-          style={{
-            height: graphHeight,
-            width: graphWidth,
-            padding: this.graphVerticalPadding + 'px ' + nodeRadius * 2 + 'px ' + '0 ' + nodeRadius * 2 + 'px',
-          }}
-        >
-          <g className="placeholder">
-            <foreignObject width={maxLabelWidth > 0 ? maxLabelWidth : 1} height="200">
-              <div className="label-body node active" />
-            </foreignObject>
-          </g>
-          {allNodes.map(
-            (node) =>
-              !node.placeholder && (
-                <g
-                  key={node.id}
-                  className={classNames({
-                    'has-status': !!node.status,
-                    active: node.isActive,
-                    highlighted: node.isHighlighted,
-                    warning: node.hasWarnings,
-                  })}
-                  transform={`translate(${node.x},${node.y})`}
-                >
-                  {node.childLinks.map((link) => (
-                    <PipelineGraphLink key={`${link.child.id}_${link.parent.name}`} link={link} x={node.x} y={node.y} />
-                  ))}
-                  <PipelineGraphNode
-                    isExecution={!!execution}
-                    labelOffsetX={labelOffsetX}
-                    labelOffsetY={labelOffsetY}
-                    maxLabelWidth={maxLabelWidth}
-                    nodeClicked={this.props.onNodeClick}
-                    highlight={this.highlight}
-                    nodeRadius={nodeRadius}
-                    node={node}
-                  />
-                </g>
-              ),
-          )}
-        </svg>
+        {this.getPipelineGraphRender()}
```

Comment:
Is there a reason for moving everything out of render into its own method? If this was one part of what was being returned it might make sense, but given this is the only thing returned by this component it's an extra layer of abstraction that I don't think is needed?

Reply:
Yeah, I agree with you! I don't think we need to make this abstraction (Putting it in the render method would make it more readable too!). Good catch! Ill make the change 👍